### PR TITLE
feat: IFS-3776 Angular 18 migration

### PR DESCRIPTION
**Pull request by mtpham-collab**

### CODE REVIEW

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 1]
```diff
@@ -86,7 +86,8 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
   let spectator: Spectator<TestComponent>;
   const createComponent = createComponentFactory({
     component: TestComponent,
-    imports: [WizardModule, RouterTestingModule, IncompleteDateModule]
+    imports: [WizardModule, IncompleteDateModule],
+    providers: [WidgetsConfigService, provideRouter([])]
```
Comment (msg-jstren) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:90]: Muss der WidgetsConfigService in providers?

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-anzeigen/objekt-anzeigen.component.spec.ts [hunk 2]
```diff
@@ -59,15 +58,6 @@ describe('Integration Tests: ObjektAnzeigenComponent', () => {
...
 
-  /**
-   * Is clicking a button
-   * @param sbutton the ID of the button who must be clicked
-   */
-  function clickButton(sbutton: string): void {
-    const button = fixture.nativeElement.querySelector(sbutton) as HTMLButtonElement;
-    button.click();
-  }
-
```
[the remark this answers is not in the document]
Comment (mtpham-collab) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-anzeigen/objekt-anzeigen.component.spec.ts:70]: The function is no longer used

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts (hunk 1 above)
Comment (mtpham-collab) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:90]: WidgetsConfigService  was deleted

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# 18.0.0
+## Breaking Changes
+- IFS-3776: Upgrade der Angularversion von 17 auf 18
+
 # 17.5.0
 ## Feature
 - IFS-4016: Integration des FluentUI-Themes und Entfernung des Bootstrap-Light-Themes im Angular-Baustein
```
Comment (msg-jstren) [inline CHANGELOG.md:7]: Muss unter Version #18.0.0 stehen

File: UPDATELOG.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
+# Update Log
```
Comment (msg-jstren) [inline UPDATELOG.md:1]:
```suggestion
# Update Log - 02.09.2024
```

**Review by msg-jstren — DISMISSED** [review]

**Review by msg-jstren — APPROVED** [review]

**Pull request merged**